cmd/gin-repod/repo.go

```diff
@@ -54,10 +54,17 @@ func (s *Server) createRepo(w http.ResponseWriter, r *http.Request) {
 
 	rid := store.RepoId{Owner: owner, Name: creat.Name}
 
-	_, ok := s.checkAccess(w, r, rid, store.NoAccess)
+	user, ok := s.checkAccess(w, r, rid, store.NoAccess)
 	if !ok {
 		return
 	}
+	// make sure routes user and token user are identical
+	if owner != user.Uid {
```

Comment:
I checked and there should actually be only one case, when user will be `nil,` while `ok` is `true.` This only happens when the authentication header is completely missing, in any other case `checkAccess` ends the request with an error. Since we want `checkAccess` under normal circumstances to continue when the header is missing to check for other stuff, I will add a `nil` check for user at the `createRepo` function level.